thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/email/filter/BaseAlertFilter.java

```diff
@@ -23,23 +28,34 @@ public void setParameters(Map<String, String> parameterSetting) {
     Class c = this.getClass();
     for (String fieldName : getPropertyNames()) {
       Double value = null;
+      String strVal = null;
       // Get user's value for the specified field
       if (parameterSetting.containsKey(fieldName)) {
-        value = Double.parseDouble(parameterSetting.get(fieldName));
+        String fieldVal = parameterSetting.get(fieldName);
+        if (NumberUtils.isNumber(fieldVal)) {
+          value = Double.parseDouble(parameterSetting.get(fieldName));
+        } else {
+          strVal = fieldVal;
+        }
       } else {
         // If user's value does not exist, try to get the default value from Class definition
         try {
-          Field field = c.getDeclaredField("DEFAULT_" + fieldName.toUpperCase());
+          Field field = c.getDeclaredField(fieldName);
```

Comment:
This change might break existing classes. This block is trying to get default value for variables when their value is not given by user. The naming convention DEFAULT_VARIABLE_NAME. If you remove the original line, then this block does the same thing at the block at lines 33-39.